client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -379,6 +362,18 @@ public void appendSetup(AppendSetup appendSetup) {
             }
         }
 
+        private void retryOnWriteFail(WireCommand wireCommand) {
+            if (state.encounteredSeal()) {
+                Retry.indefinitelyWithExpBackoff(retrySchedule.getInitialMillis(), retrySchedule.getMultiplier(),
+                        retrySchedule.getMaxDelay(),
+                        t -> log.error("Error while invoking SegmentSealed call back for writer: {}", writerId, t))
+                        .runInExecutor(() -> {
+                            log.debug("Invoking SealedSegment call back for {} on writer {}", wireCommand, writerId);
+                            callBackForSealed.accept(Segment.fromScopedName(getSegmentName()));
+                        }, connectionFactory.getInternalExecutor());
```

Comment:
This is undoing your recent change to delay release until after invoking the callback.

Reply:
done, merged with the latest to include those changes.